backend/services/email_service.py

```
@@ -0,0 +1,350 @@
+        """Initialize email service with SMTP configuration."""
+        # SMTP configuration loaded from environment variables
+        # This ensures sensitive credentials are not hardcoded
+        self.smtp_host = settings.smtp_host
+        self.smtp_port = settings.smtp_port
+        self.smtp_user = settings.smtp_user
+        self.smtp_password = settings.smtp_password
+        self.from_email = settings.from_email
+        
+        # Initialize Jinja2 environment for email templates
+        # BaseLoader allows us to define templates as strings
+        self.template_env = Environment(loader=BaseLoader())
```

Ruff (0.13.2)

By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)